### DISCUSSION TITLE
"Computing issues" - benchmarks

### DISCUSSION THREAD

**Discussion by nurbs999** [sn4k3/UVtools#935] [General]
Hey folks,

I was thinking about upgrading my rig and did some benchmarks with my CPUs. Unfortunately UVTools doesn't scale well with CPU IPC or core counts, I guess. So I checked the CPU utilisation while the analyse process is running.
Computing issues runs on all cores, which is great. But all other steps, like detecting resin traps, are running on just a view cores and with just about 40% core utilisation.
All my machines have 64GB of memory, which should be enough. UVTools reports a usage of around 15-16GB.
I uploaded my benchmark file here: 
https://www.printables.com/model/1019269-uvtools-benchmark-file 
and my current results here:
https://www.reddit.com/r/resinprinting/comments/1fba7ik/uvtools_cpu_benchmarks/?rdt=42994

What do you think? Am I missing something? Is something else "bottlenecking" the tool?

**Comment by sn4k3** [comment]
If you upgrade CPU without clean install you must check UVtools settings for the max degree of paralelism. There is a change you keep a positive number there which limits your cores to same amount as before. Upgrading CPU will benefit as all load is made on CPU.

Resin traps and suction cups can relief a bit of load as it need to run sequential on some stages and not brute force your cores with parallel work.

Also note UVtools have a benchmark tool under Help menu

**Comment by nurbs999** [comment]
Thanks for the hint!
I checked the setting and tried -2 and my current core count 32. Both runs need 7m7s on a 9950x (32 threads) compared to 9m32s on a 5800x (16 threads). This is windows, btw. I just tried linux, et voila: 4m53s on the 9950x. This is 45% faster than windows.

**Reply by sn4k3** [reply]
9950x is very recent, I have seen reports of faster usage under linux. It may be fixed soon on Windows.
Beside CPU a faster RAM will also help.
With the increase of image resolution, it will increase times by a large factor.

**Reply by nurbs999** [reply]
Thats true. Do you think a difference of 45% is all related to the OS/scheduler?

**Reply by sn4k3** [reply]
Is that with a fresh install of windows? Maybe wait for new 24h2 update, for sure it will improve.

**Reply by nurbs999** [reply]
It's windows 10, and I tested both, with and without a fresh installation. Didn't make any difference.

**Reply by nurbs999** [reply]
I installed windows 11 23H2 ran the test file:

Ryzen 9 9950X Linux	        04m53s
Ryzen 9 9950X Windows 11	06m05s
Ryzen 9 9950X Windows 10	07m07s

Maybe you are right. I'll wait for the 24H2 update and retest.
